File: app/src/main/java/me/bkkn/domain/repository/Notes.java
package me.bkkn.domain.repository;

import java.util.List;

import me.bkkn.domain.entity.Note;

public interface Notes {
    List<Note>getNotes();

    void deleteNote(Note note);

    void changeTitle(Note note, String newTitle);

    void editNote(Note note, String newTitle, String newContent);

    void addNewNote();

    void addNewNote(String title, String content);

    default int index(Note note)
    {
        return getNotes().indexOf(note);
    }
}

File: app/src/main/java/me/bkkn/domain/repository/Notes.kt
package me.bkkn.domain.repository;

import java.util.List;

import me.bkkn.domain.entity.Note;

public interface Notes {
    List<Note> getlist();

    void deleteNote(Note note);

    void changeTitle(Note note, String newTitle);

    void editNote(Note note, String newTitle, String newContent);

    void addNewNote();

    default int index(Note note)
    {
        return getlist().indexOf(note);
    }
}
